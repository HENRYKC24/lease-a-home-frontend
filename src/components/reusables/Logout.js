import React from 'react';

const Logout = () => {
  console.log(883);
  const handleLogout = () => {
    const authKey = localStorage.getItem('authKey');
    console.log(authKey, '<<<auth key>>>');
  };
  return (
    <button
      onClick={handleLogout}
      style={{ backgroundColor: colorScheme.blue }}
      type="button"
      className={logoutBtn}
    >
      Logout
    </button>
  );
};

export default Logout;
