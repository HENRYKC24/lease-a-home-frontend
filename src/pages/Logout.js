import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import colorScheme from '../colorScheme';
import style from './signup.module.css';
import { hitAPIWithLogoutDetails } from '../redux/user/user';

const Logout = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const user = useSelector((state) => state.user);
  const { loggedIn } = user;

  const [show, setShow] = useState(true);
  const [isLoggedIn, setIsLoggedIn] = useState(false);

  const {
    overlay,
    cancel,
    yes,
    btnWrapper,
    cancelBtn,
    yesBtn,
    overlayInner,
    areYouSure,
  } = style;

  const handleLogout = () => {
    const userAuth = JSON.parse(localStorage.getItem('userAuth'));
    dispatch(hitAPIWithLogoutDetails({ userAuth }));
    setIsLoggedIn(() => true);
  };

  const handleCancel = () => {
    setShow(() => false);
    navigate('/', { replace: true });
  };

  useEffect(() => {
    if (loggedIn === 'out' && isLoggedIn) {
      setShow(() => false);
      navigate('/', { replace: true });
    }
  }, [user, setIsLoggedIn]);

  useEffect(() => {
    window.scrollTo(0, 0);
    document.body.classList.add('stop-scrolling');
    return () => {
      document.body.classList.remove('stop-scrolling');
    };
  }, []);

  return show ? (
    <div className={overlay}>
      <div className={overlayInner}>
        <p className={areYouSure}>Are you sure you want to log out?</p>
        <div className={btnWrapper}>
          <div className={cancel}>
            <button
              onClick={handleCancel}
              style={{ backgroundColor: colorScheme.blue }}
              type="submit"
              className={cancelBtn}
            >
              Cancel
            </button>
          </div>

          <div className={yes}>
            <button
              onClick={handleLogout}
              style={{ backgroundColor: colorScheme.blue }}
              type="submit"
              className={yesBtn}
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </div>
  ) : null;
};

export default Logout;
