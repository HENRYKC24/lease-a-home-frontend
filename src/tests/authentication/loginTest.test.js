import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Logout from '../../pages/Logout';
import store from '../../redux/configureStore';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

it('Login renders the UI as expected', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Logout />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
