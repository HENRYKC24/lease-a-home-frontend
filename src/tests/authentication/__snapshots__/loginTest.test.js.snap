// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders the UI as expected 1`] = `
<div
  className="overlay"
>
  <div
    className="overlayInner"
  >
    <p
      className="areYouSure"
    >
      Are you sure you want to log out?
    </p>
    <div
      className="btnWrapper"
    >
      <div
        className="cancel"
      >
        <button
          className="cancelBtn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#10BBB5",
            }
          }
          type="submit"
        >
          Cancel
        </button>
      </div>
      <div
        className="yes"
      >
        <button
          className="yesBtn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#10BBB5",
            }
          }
          type="submit"
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;
